libtransmission/blocklist.cc

```diff
@@ -49,10 +50,67 @@ void BlocklistFile::ensureLoaded() const
         return;
     }
 
-    auto range = IPv4Range{};
-    while (in.read(reinterpret_cast<char*>(&range), sizeof(range)))
+    auto file_size = std::filesystem::file_size(filename_);
+    auto zeroes_count = 0;
+    auto max_zeroes = 0;
+
+    if (file_size >= 40)
```

Comment:
Let's avoid hardcoding `40`

```suggestion
    static auto constexpr RangeSize = sizeof(AddressRange);
    if (file_size >= RangeSize)
```

(also the `40` instances below)